thrust/system/cuda/detail/scan_by_key.h

```diff
@@ -512,7 +512,7 @@ namespace __scan_by_key {
             inequality_op(equality_op_),
             scan_op(scan_op_)
       {
-        int  tile_idx      = blockIdx.x;
+        Size tile_idx      = blockIdx.x;
```

Comment:
Can you switch this back to using a `static_cast<Size>` in the `tile_base` calculation and leave `tile_idx` as an `int`?

`tile_idx` is passed to `consume_tile`, which expects it to be an `int`, and this change will introduce truncation warnings for 8-byte `Size` types.

Reply:
Ah this is my fault. We should actually we use `Size` for `tile_idx` all the way down, I think. It's conceivable `tile_idx` can rise higher than intmax. In my experiments `ITEMS_PER_TILE was `9 * 256`, meaning `1 >> 43` number of inputs would overflow.

Do you think having `tile_idx` as `Size` would be problematic?

Reply:
I just realized `tile_idx` this goes all the way down to `TilePrefixCallbackOp`, which accepts `int` for `tile_idx` in the constructor, so making a change to `tile_idx`'s type would require changes to all single-pass scan algorithms.

Reply:
A block's x dimension must always fit in an `int`, so it's best to leave `tile_idx` as-is. If we needed more tiles, it'd need to be handled at a higher level of the implementation.